### PR TITLE
Enforce centralized sequencer in p2p block gossiping

### LINKED ISSUE #1404: Remove redundant boiler plate from tests
[Here](https://github.com/rollkit/rollkit/blob/main/node/full_node_integration_test.go#L43) we setup a new node from scratch. It can be cleaned up by using [this](https://github.com/rollkit/rollkit/blob/main/node/node_test.go#L67).

### LINKED ISSUE #1417: Validate blocks in BlockStoreRetrieveLoop before storing them
Consists of informal DA audit issues:

- _Blocks arriving out of order could halt the chain_
- _The authenticity of blocks received through the p2p layer is
not checked_
- _checkCentralizedSequencer does not check that the block
was created by the trusted sequencer_

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Resolves #1417, resolves #1404

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
	- Enhanced block validation to reject certain blocks early in the process.
	- Implemented centralized sequencer validation to ensure block sequencing integrity.

- **Bug Fixes**
	- Adjusted the initialization of nodes in test suites to align with updated genesis document and signing key configurations.
	- Updated error handling in block management to provide more informative feedback.

- **Refactor**
	- Streamlined block manager initialization with improved error handling.
	- Updated test cases to reflect new validator set and signing key initialization methods.
	- Refined node test initialization to include signing key handling and error propagation.

- **Documentation**
	- Added new error messages to improve clarity on validation failures.

- **Tests**
	- Enhanced integration and unit tests to cover new validation checks and error conditions.

### CODE REVIEW

File: node/full_node_integration_test.go [hunk 1]
```diff
@@ -40,78 +37,6 @@ func prepareProposalResponse(_ context.Context, req *abci.RequestPrepareProposal
 	}, nil
 }
 
-func TestCentralizedSequencer(t *testing.T) {
```
Comment (S1nus) [inline node/full_node_integration_test.go:43]: why remove this test?

**Review by S1nus — DISMISSED** [review]

**Review by S1nus — APPROVED** [review]

File: node/node_test.go [hunk 2]
```diff
@@ -35,15 +35,15 @@ func initializeAndStartLightNode(ctx context.Context, t *testing.T) *LightNode {
...
+	node, _ := setupTestNode(ctx, t, nodeType)
 	require.False(t, node.IsRunning())
 	err := node.Start()
 	require.NoError(t, err)
 	require.True(t, node.IsRunning())
 	return node
 }
 
-func newTestNode(ctx context.Context, t *testing.T, nodeType string) (Node, error) {
+func newTestNode(ctx context.Context, t *testing.T, nodeType string) (Node, error, ed25519.PrivKey) {
```
Comment (MSevey) [inline node/node_test.go:46]:
```suggestion
func newTestNode(ctx context.Context, t *testing.T, nodeType string) (Node, ed25519.PrivKey, error) {
```
we should keep the convention of ending with an error.

**Review by MSevey — APPROVED** [review]
minor update about return values.

File: node/node_test.go (hunk 2 above)
Comment (Manav-Aggarwal) [reply node/node_test.go]: fixed

**Review by MSevey — APPROVED** [review]

File: types/signed_header.go [hunk 3]
```diff
@@ -137,9 +127,19 @@ func (sh *SignedHeader) ValidateBasic() error {
 		return err
 	}
 
-	// Rollkit vA uses a centralized sequencer.
-	if err := sh.checkCentralizedSequencer(); err != nil {
-		return err
+	// Rollkit vA uses a centralized sequencer, so there should only be one validator
```
Comment (S1nus) [inline types/signed_header.go:130]: I'm fine with this, but I was previously asked to separate this into its own function

**Review by S1nus — APPROVED** [review]

File: types/signed_header.go (hunk 3 above)
Comment (Manav-Aggarwal) [reply types/signed_header.go:130]: Yes, I understand and have that context too. I think it's fine to roll with this.

**Pull request merged**